File: Anul2/Semestrul_1/Metode avansate de programare/Laboratoare/Lab2/lab2_schelet/sem1_2/decorator/DelayTaskRunner.java
package sem1_2.decorator;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class DelayTaskRunner extends AbstractTaskRunner{

    private static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ofPattern("HH:mm");

    public DelayTaskRunner(TaskRunner taskRunner) {
        super(taskRunner);
    }

    @Override
    public void executeOneTask() {
        try {
            super.executeOneTask();
            System.out.println("Task executed at: " + LocalDateTime.now().format(TIME_FORMATTER));
            Thread.sleep(3000);

        } catch (InterruptedException e) {
            e.printStackTrace();
        }


    }
}

File: Anul2/Semestrul_1/Metode avansate de programare/Laboratoare/Lab9/ConsoleApp1/decorator/DelayTaskRunner.cs
namespace ConsoleApp1.decorator;

public class DelayTaskRunner : AbstractTaskRunner
{
    public DelayTaskRunner(ITaskRunner taskRunner) : base(taskRunner) {}
    public override void ExecuteOneTask() {
        try {
            base.ExecuteOneTask();
            System.Console.WriteLine("Task executed at: " + DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss"));
            Thread.Sleep(3000);

        } catch (Exception e)
        {
           System.Console.WriteLine(e);
        }
    }
}